public/pages/CreateComposableTemplate/containers/TemplateDetail/TemplateDetail.tsx

```diff
@@ -209,7 +210,7 @@ const TemplateDetail = (props: TemplateDetailProps, ref: Ref<IComponentTemplateD
       )}
       <DefineTemplate {...subCompontentProps} />
       <EuiSpacer />
-      <IndexAlias {...subCompontentProps} />
+      <IndexAlias key={props.dataSourceId} {...subCompontentProps} />
```

Comment:
A comment here will be nice, 

```
add key to force unmount the component to let it retrieve the alias list from current data source.
```

something like that.

Reply:
Agree, didn't add because adding comment in JSX looks bad... Added as part of next commit.